src/components/language/language.service.ts

```diff
@@ -49,18 +49,15 @@ export class LanguageService {
     private readonly projectService: ProjectService & {},
     @Inject(forwardRef(() => EngagementService))
     private readonly engagementService: EngagementService & {},
-    private readonly authorizationService: AuthorizationService,
+    private readonly privileges: Privileges,
     private readonly repo: LanguageRepository,
     @Logger('language:service') private readonly logger: ILogger,
   ) {}
 
   async create(input: CreateLanguage, session: Session): Promise<Language> {
-    await this.authorizationService.checkPower(Powers.CreateLanguage, session);
+    this.privileges.for(session, Language).verifyCan('create');
 
-    await this.authorizationService.checkPower(
-      Powers.CreateEthnologueLanguage,
-      session,
-    );
+    this.privileges.for(session, EthnologueLanguage).verifyCan('create');
```

Comment:
Let's drop this. EthnologueLanguage's are always created as sub-objects of Language. They are never directly user created. The first check is sufficient.
```suggestion
```